openstack/compute/v2/extensions/keypairs/requests.go

```diff
@@ -76,14 +80,28 @@ func Create(client *gophercloud.ServiceClient, opts CreateOptsBuilder) (r Create
 
 // Get returns public data about a previously uploaded KeyPair.
 func Get(client *gophercloud.ServiceClient, name string) (r GetResult) {
-	resp, err := client.Get(getURL(client, name), &r.Body, nil)
+	r = GetWithUserID(client, name, "")
+	return
+}
+
+// GetWithUserID returns public data about a previously uploaded KeyPair, owned by a specific userID.
+// If the userID is an empty string, it searches the KeyPairs of the current user.
+func GetWithUserID(client *gophercloud.ServiceClient, name string, userID string) (r GetResult) {
```

Comment:
Instead of passing in `userID` in the function parameters, create a `GetOpts` struct that contains a `UserID` field. See here for an example: https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/extensions/limits/requests.go

The reason this is done is because the user ID is a query parameter of the URL and not part of the URL itself.